services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/Utils.kt

```diff
@@ -43,13 +53,18 @@ fun BusinessId.isValidBusinessId(): Boolean {
         .sumOf { (num, multiplier) -> num * multiplier }
         .mod(11)
         .let { remainder ->
-            val check = substringAfter("-").toInt()
-            val result = 11 - remainder
+            val calculatedCheck = if (remainder == 0) 0 else 11 - remainder
 
-            return when {
-                remainder == 1 -> false.also { logger.warn { "Remainder not valid." } }
-                (remainder == 0 && 0 == check) || result == check -> true
-                else -> false.also { logger.warn { "Invalid businessId." } }
+            if (remainder == 1) {
+                logger.warn { "Remainder not valid." }
+                return false
```

Comment:
I think it would be clearer if calculating the check digit would be clearly separated from checking the check digit.

Like,
```kotlin
    val calculatedCheck =
        sequenceNumber
            .map { it.digitToInt() }
            .zip(listOf(7, 9, 10, 5, 8, 4, 2))
            .sumOf { (num, multiplier) -> num * multiplier }
            .mod(11)
            .let { remainder ->
                when (remainder) {
                    1 -> {
                        logger.warn { "Remainder not valid." }
                        return false
                    }
                    0 -> 0
                    else -> 11 - remainder
                }
            }
```
After this, checking that the given and calculated check digits are the same can be done in separate, clear step:
```kotlin
    return if (calculatedCheck == checkDigit.toInt()) {
        true
    } else {
        logger.warn { "Check digit doesn't match." }
        false
    }
```

Reply:
Added this with slight modifications.